scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java

```diff
@@ -1046,6 +1107,15 @@ protected TaskLauncherInitializer getDefaultTaskLauncherInitializer(InternalJob
         Map<String, String> gInfo = getGenericInformationOverridden(job);
         tli.setGenericInformation(gInfo);
 
+        ForkEnvironment environment = getForkEnvironment();
+        if (environment != null) {
+            Script environmentScript = environment.getEnvScript();
+            if (environmentScript != null) {
+                if (!isScriptAuthorized(getId(), environmentScript)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1066)